spec/features/openid_connect/openid_connect_spec.rb

```diff
@@ -140,6 +140,28 @@
     expect(page).to have_content(t('headings.sign_in_without_sp'))
   end
 
+  it 'returns verified_at in an ial1 session if requested' do
+    profile = create(:profile, :active, :verified,
+                     pii: { first_name: 'John', ssn: '111223333' })
+    user = profile.user
+
+    token_response = sign_in_get_token_response(
```

Comment:
This test is failing and my brain isn't working so I'm leaving this here for the day.

This *should* have been a simple extraction, but this helper method doesn't work here, I'd love some pointers on what implied context or pre-step I'm missing.... I basically wanted:

- To be able to get the raw `token` response so I can grab `access_token` (other specs just need `id_token`)
- To be able to supply a user that has a verified profile (so it actually has a `verified_at` value)

It fails inside the helper because where the code expects a redirected URL (that it can grab params from) instead it's on `sign_up/completed` (which doesn't make sense since it should be sign *in* not *up* ?)

Reply:
Sign up completed is the page where you consent to share data with an SP. You land there if you have never signed into an SP before or you are requesting a scope you have not in the past. The way it determines what attributes you have requested is based on the `verified_attributes` column on the identity.

The `sign_in_get_token_response` helper sets `email` in the verified attributes here: https://github.com/18F/identity-idp/blob/785556765cfd156c6a224b3728555e00e78fa9c0/spec/features/openid_connect/openid_connect_spec.rb#L465

Since `profile:verified_at` is not saved there the authorization controller sees that you have new scopes to authorize and sends you to sign up completed to authorize them.

Reply:
So you can either add the scope to the verified attributes or click "Continue" on the consent screen to get past this point

Reply:
Some references:

Here is where the redirect to sign up completed happens in the authorization controller: https://github.com/18F/identity-idp/blob/master/app/controllers/openid_connect/authorization_controller.rb#L24

Here is where the authorization controller looks to check whether it should redirect: https://github.com/18F/identity-idp/blob/master/app/controllers/concerns/verify_sp_attributes_concern.rb#L6-L14

The rest of the logic to determine if there are new scopes lives in that last concern.

Reply:
Thanks for all this!